src/app/clusters/ota-requestor/ExtendedOTARequestorDriver.cpp

```diff
@@ -68,9 +68,16 @@ void ExtendedOTARequestorDriver::PollUserConsentState()
 CHIP_ERROR ExtendedOTARequestorDriver::GetUserConsentSubject(chip::ota::UserConsentSubject & subject,
                                                              const UpdateDescription & update)
 {
-    // mLastUsedProvider has the provider fabric index and endpoint id
-    subject.fabricIndex        = mLastUsedProvider.fabricIndex;
-    subject.providerEndpointId = mLastUsedProvider.endpoint;
+    if (mLastUsedProvider.HasValue())
+    {
+        // mLastUsedProvider has the provider fabric index and endpoint id
+        subject.fabricIndex        = mLastUsedProvider.Value().fabricIndex;
+        subject.providerEndpointId = mLastUsedProvider.Value().endpoint;
+    }
+    else
+    {
+        ChipLogError(SoftwareUpdate, "ExtendedOTARequestorDriver::GetUserConsentSubject: mLastProvider is empty");
```

Comment:
Is the user of this function happy with fabricIndex and providerEndpointId not being initialized? Shouldn't we simply error in such a case?

Reply:
Agreed- unless there's some convoluted way where the mLastUsedProvider was not set before sending the QueryImage out.
I see that OTARequestor::TriggerImmediateQuery() does not set the mLastUsedProvider, maybe there are others, that should be fixed?

Reply:
Yes, TriggerImmediateQuery() should set mLastUsedProvider, it's a bug. It should be set every time SetCurrentProviderLocation() is called.

Reply:
https://github.com/project-chip/connectedhomeip/issues/15691 will add another safeguard in this area